src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -589,5 +631,74 @@ private static string GetFullPath(K8SConfiguration configuration, string path)
                 return Path.Combine(Path.GetDirectoryName(configuration.FileName), path);
             }
         }
+
+        private static void MergeKubeConfig(K8SConfiguration basek8SConfig, K8SConfiguration mergek8SConfig)
+        {
+            // For scalar values, prefer local values 
+            basek8SConfig.CurrentContext = basek8SConfig.CurrentContext ?? mergek8SConfig.CurrentContext;
+            basek8SConfig.FileName = basek8SConfig.FileName ?? mergek8SConfig.FileName;
+            basek8SConfig.Kind = basek8SConfig.Kind ?? mergek8SConfig.Kind;
+
+            // For Dictionaries, prefer first if not present in second.
+            if (mergek8SConfig.Preferences != null && mergek8SConfig.Preferences.Count > 0)
+            {
+                foreach (var preference in mergek8SConfig.Preferences)
+                {
+                    if (basek8SConfig.Preferences?.ContainsKey(preference.Key) == false)
+                    {
+                        basek8SConfig.Preferences[preference.Key] = preference.Value;
+                    }
+                }
+            }
+
+            if (mergek8SConfig.Extensions != null && mergek8SConfig.Extensions.Count > 0)
+            {
+                foreach (var extension in mergek8SConfig.Extensions)
+                {
+                    if (basek8SConfig.Extensions?.ContainsKey(extension.Key) == false)
+                    {
+                        basek8SConfig.Extensions[extension.Key] = extension.Value;
+                    }
+                }
+            }
+
+            if (mergek8SConfig.Clusters != null && mergek8SConfig.Clusters.Count() > 0)
+            {
+                var clusters = new List<Cluster>();
```

Comment:
If you're going to make a temp data structure, it might be simpler to use a dictionary where the key is the cluster name.